src/main/java/uk/gov/hmcts/reform/sscs/service/ccdupdate/HearingsJourneyService.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.hmcts.reform.sscs.service.ccdupdate;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.exception.GetCaseException;
+import uk.gov.hmcts.reform.sscs.exception.UpdateCaseException;
+import uk.gov.hmcts.reform.sscs.model.messaging.HmcMessage;
+import uk.gov.hmcts.reform.sscs.model.single.hearing.HearingGetResponse;
+import uk.gov.hmcts.reform.sscs.service.CcdCaseService;
+import uk.gov.hmcts.reform.sscs.service.HmcHearingService;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.EventType.HEARING_BOOKED;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.EventType.UPDATE_CASE_ONLY;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class HearingsJourneyService {
+
+    private static final String CREATED_SUMMARY = "SSCS - new case sent to HMC";
+    private static final String CREATED_DESC = "SSCS - new case sent to HMC";
+    private static final String UPDATED_SUMMARY = "SSCS - updated case sent to HMC";
+    private static final String UPDATED_DESC = "SSCS - updated case sent to HMC";
+    private static final String CANCELLED_SUMMARY = "SSCS - case cancelled";
+    private static final String CANCELLED_DESC = "SSCS - case cancelled";
+    private static final String EXCEPTION_SUMMARY = "SSCS - exception occurred";
+    private static final String EXCEPTION_DESC = "SSCS - exception occurred";
+
+    private final HmcHearingService hmcHearingService;
+    private final CcdCaseService ccdCaseService;
+    private final CcdStateUpdateService ccdStateUpdateService;
+    private final CcdLocationUpdateService locationUpdateService;
+
+    public void process(HmcMessage hmcMessage) throws GetCaseException, UpdateCaseException {
+        validateHmcMessage(hmcMessage);
```

Comment:
I don't agree with this method of validation.  If it's the pattern used, I'll live with it, but checking for null's then throwing a NullPointerException is a bad method of validation.  Instead, validate and throw back a checked exception that can be managed higher up appropriately.  Also, should we not be checking the quality of the values beyond just null checks?
Consider what would be using these values and what format latter code expects the values to be in.
Treat data coming in as if it could be garbage and check for that.

Reply:
will do